source/Playnite/GamesEditor.cs

```diff
@@ -467,6 +467,140 @@ public void OpenGameLocation(Game game)
+                    }
+                }
+            }
+            catch (Exception e) when (!PlayniteEnvironment.ThrowAllErrors)
+            {
+                logger.Error(e, $"Failed to get InstallSize from game {game.Name} with install dir {game.InstallDirectory}");
+                Dialogs.ShowMessage(
+                    string.Format(resources.GetString("LOCCalculateGameSizeError"), e.Message),
+                    resources.GetString("LOCGameError"),
+                    MessageBoxButton.OK, MessageBoxImage.Error);
+                return;
+            }
+        }
+
+        public void CalculateGameSizeWithDialog(Game game, bool onlyIfDataMissing, bool updateGame = true)
+        {
+            var textTitle = string.Format(ResourceProvider.GetString("LOCCalculatingInstallSizeOfGameMessage"), game.Name);
+            Dialogs.ActivateGlobalProgress((a) =>
+            {
+                CalculateGameSize(game, onlyIfDataMissing, updateGame);
+            }, new GlobalProgressOptions(textTitle, false) { IsIndeterminate = true });
+        }
+
+        public void CalculateGamesSize(List<Game> games, bool onlyIfDataMissing)
+        {
+            using (Database.BufferedUpdate())
+            {
+                foreach (var game in games)
+                {
+                    CalculateGameSize(game, onlyIfDataMissing);
+                }
+            }
+        }
+
+        public void CalculateGamesSizeWithDialog(List<Game> games, bool onlyIfDataMissing)
+        {
+            var textTitle = ResourceProvider.GetString("LOCCalculatingInstallSizeMessage");
+            Dialogs.ActivateGlobalProgress((a) =>
+            {
+                a.ProgressMaxValue = games.Count();
+                using (Database.BufferedUpdate())
+                {
+                    foreach (var game in games)
+                    {
+                        if (a.CancelToken.IsCancellationRequested)
+                        {
+                            break;
+                        }
+
+                        a.CurrentProgressValue++;
+                        a.Text = $"{textTitle}\n\n{game.Name}\n{a.CurrentProgressValue}/{a.ProgressMaxValue}";
+                        CalculateGameSize(game, onlyIfDataMissing);
+                    }
+                }
+            }, new GlobalProgressOptions(textTitle, true) { IsIndeterminate = false });
+        }
+
+        private long? GetGameInstallDirSize(Game game)
+        {
+            if (game.InstallDirectory.IsNullOrEmpty() || !FileSystem.DirectoryExists(game.InstallDirectory))
+            {
+                return null;
+            }
+
+            if (AppSettings.InstallSizeScanUseSizeOnDisk)
+            {
+                return FileSystem.GetDirectorySizeOnDisk(game.InstallDirectory);
+            }
+            else
+            {
+                return FileSystem.GetDirectorySize(game.InstallDirectory);
+            }
+        }
+
+        private long? GetGameRomSizes(Game game)

```

Comment:
This will probably fail with ROMs that have relative paths assigned with variables pointing to {EmulatorDir} via install folder.

Reply:
Fixed as part of https://github.com/JosefNemec/Playnite/pull/3017/commits/14ebf7d58bdedc60f211eaaec652fb4e052bd280